buffer/src/main/java/io/netty/buffer/api/adaptor/ByteBufBuffer.java

```diff
@@ -0,0 +1,1312 @@
+    }
+
+    @Override
+    protected Owned<ByteBufBuffer> prepareSend() {
+        final ByteBuf delegate = this.delegate;
+        return new Owned<ByteBufBuffer>() {
+            @Override
+            public ByteBufBuffer transferOwnership(Drop<ByteBufBuffer> drop) {
+                return new ByteBufBuffer(control, delegate, drop);
+            }
+        };
+    }
+
+    @Override
+    protected void makeInaccessible() {
+        super.makeInaccessible();
+        delegate = (ByteBuf) CLOSED_BYTEBUF;
+    }
+
+    @Override
+    public int capacity() {
+        return delegate.capacity();
+    }
+
+    @Override
+    public int readerOffset() {
+        return delegate.readerIndex();
+    }
+
+    @Override
+    public void skipReadable(int delta) {
+        delegate.readerIndex(delegate.readerIndex() + delta);
+    }
+
+    @Override
+    public Buffer readerOffset(int offset) {
+        delegate.readerIndex(offset);
+        return this;
+    }
+
+    @Override
+    public int writerOffset() {
+        return delegate.writerIndex();
+    }
+
+    @Override
+    public void skipWritable(int delta) {
+        delegate.writerIndex(delegate.writerIndex() + delta);
+    }
+
+    @Override
+    public Buffer writerOffset(int offset) {
+        if (readOnly()) {
+            throw bufferIsReadOnly(this);
+        }
+        delegate.writerIndex(offset);
+        return this;
+    }
+
+    @Override
+    public int readableBytes() {
+        return delegate.readableBytes();
+    }
+
+    @Override
+    public int writableBytes() {
+        return delegate.capacity() - delegate.writerIndex();
+    }
+
+    @Override
+    public Buffer fill(byte value) {
+        if (!isAccessible()) {
+            throw attachTrace(bufferIsClosed(this));
+        }
+        if (readOnly()) {
+            throw bufferIsReadOnly(this);
+        }
+        int capacity = capacity();
+        for (int i = 0; i < capacity; i++) {
+            delegate.setByte(i, value);
+        }
+        return this;
+    }
+
+    @Override
+    public Buffer makeReadOnly() {
+        delegate = delegate.asReadOnly();
+        return this;
+    }
+
+    @Override
+    public boolean readOnly() {
+        return delegate.isReadOnly();
+    }
+
+    @Override
+    public void copyInto(int srcPos, byte[] dest, int destPos, int length) {
+        if (!isAccessible()) {
+            throw attachTrace(bufferIsClosed(this));
+        }
+        delegate.getBytes(srcPos, dest, destPos, length);
+    }
+
+    @Override
+    public void copyInto(int srcPos, ByteBuffer dest, int destPos, int length) {
+        if (dest.isReadOnly()) {
+            throw new ReadOnlyBufferException();
+        }
+        dest = dest.duplicate().clear();
+        for (int i = 0; i < length; i++) {
+            dest.put(destPos + i, getByte(srcPos + i));
+        }
+    }
+
+    @Override
+    public void copyInto(int srcPos, Buffer dest, int destPos, int length) {
+        if (dest.readOnly()) {
```

Comment:
imho we should always check for accessible first. 
```suggestion
        if (!isAccessible()) {
            throw attachTrace(bufferIsClosed(this));
        }
        if (dest.readOnly()) {
```